### ISSUE TITLE
modResource->getTVValue/setTVValue is_string needs changed to is_numeric

### ISSUE THREAD

**Issue by wshawn** [modxcms/revolution#12542]
Based on this link: http://stackoverflow.com/questions/14052207/modx-revo-settvvalue-returns-old-data/31909888#31909888

I tested out the following scenario:

``` php
$changeit = 10;
//$changeit= 'somename';
// $changeit='10';

$r = $modx->getObject('modResource', 177);
$tvs = $r->getTemplateVars();
 if ($tvs) {
        foreach ($tvs as $object){
            if (is_object($object) && $object instanceof modTemplateVar){
               print_r($object->toArray());
            }
        }
    }

$r->setTVValue($changeit,'99999fhfhg');
echo '<hr>'. $r->getTVValue($changeit);
$r->setTVValue($changeit,'888885454564');
echo '<hr>'. $r->getTVValue($changeit);
$r->setTVValue($changeit,'123456789');
echo '<hr>'. $r->getTVValue($changeit);
```

If the numeric 10 is sent to the two functions it is evaluated as a non string, but due to is_string if '10' is sent it as a string and thereby becomes the name of the TV instead of the id.

I suggest changing

``` php
$byName = false;
        if (is_string($pk)) {
            $byName = true;
        }
```

To:

``` php
       $byName = (is_numeric($pk)) ? false : true;
```

Affects:
#11486
#7453

**Comment by pbowyer** [comment]
Claiming this for myself :)

**Comment by Mark-H** [comment]
Fixed in 2.5.6
